mempool/v0/clist_mempool.go

```diff
@@ -236,7 +236,13 @@ func (mem *CListMempool) CheckTx(
 		return err
 	}
 
-	if !mem.cache.Push(tx) { // if the transaction already exists in the cache
+	originalTx := tx
+	bTx, isBlob := types.UnmarshalBlobTx(tx)
+	if isBlob {
+		tx = bTx.Tx
+	}
```

Comment:
What's actually the reason for doing this changed? The only difference I see here is that the `TxKey` is now the hash of `BlobTx.tx` rather than `BlobTx`. Is that the intention or are there other reasons? This seems to add a lot of complexity.

Reply:
apologies, this should have been documented more in an ADR, but is currently being added to an in progress. I updated the PR description to have a summary of that in #892 

Tendermint was designed around the hash of the transactions that users submitting, and the hash of the tx included on chain, be the same. For us, this is not the case, as a user submits a single transaction that contains the encoded sdk.Tx that will be included onchain along with the blob, but these things are separated before they are included in the block. Similar to what we do currently, where we keep track of each transaction with the original transaction's hash, this PR uses the hash of the encoded sdk.Tx that gets included on chain. This way, after the transaction gets included, we can hash it without having to lookup the relevant blob.

Reply:
So a user would query if their transaction has been committed based on the hash of just the `MsgPayForBlob` not the hash of the `BlobTx`?

Reply:
Have you thought about simply modifying `func (tx Tx) Key() TxKey`?